src/btree/bt_cursor.c

```diff
@@ -101,17 +101,15 @@ __cursor_page_pinned(WT_CURSOR_BTREE *cbt, bool search_operation)
         return (false);
 
     /*
-     * If we are doing an update, we need a page with history, release the page so we get it again
```

Comment:
I know it's kind of a tangent, but can you add an assert that the cbt->ref->state is only in WT_REF_MEM or WT_REF_LIMBO here? It may be possible that it's a WT_REF_DELETED as well?

It took me a while to think through that the cursor has a hazard pointer, so the page should be in one of those states.

Reply:
> add an assert that the cbt->ref->state is only in WT_REF_MEM or WT_REF_LIMBO here? It may be possible that it's a WT_REF_DELETED as well?

That makes sense to me -- I don't think it can be `WT_REF_DELETED`, so it would be good to know if that's ever the case.

Pushed in 509bab7c8.